lisa-utils/src/main/scala/lisa/utils/Serialization.scala

```diff
@@ -0,0 +1,721 @@
+package lisa.utils
+
+import lisa.utils.K.*
+import lisa.utils.ProofsShrink.*
+
+import java.io._
+import scala.collection.mutable.{Map => MutMap}
+
+object Serialization {
+
+  // Define codes for the various proof steps
+  inline def restate: Byte = 0
+  inline def restateTrue: Byte = 1
+  inline def hypothesis: Byte = 2
+  inline def cut: Byte = 3
+  inline def leftAnd: Byte = 4
+  inline def leftOr: Byte = 5
+  inline def leftImplies: Byte = 6
+  inline def leftIff: Byte = 7
+  inline def leftNot: Byte = 8
+  inline def leftForall: Byte = 9
+  inline def leftExists: Byte = 10
+  inline def leftExistsOne: Byte = 11
+  inline def rightAnd: Byte = 12
+  inline def rightOr: Byte = 13
+  inline def rightImplies: Byte = 14
+  inline def rightIff: Byte = 15
+  inline def rightNot: Byte = 16
+  inline def rightForall: Byte = 17
+  inline def rightExists: Byte = 18
+  inline def rightExistsOne: Byte = 19
+  inline def weakening: Byte = 20
+  inline def leftRefl: Byte = 21
+  inline def rightRefl: Byte = 22
+  inline def leftSubstEq: Byte = 23
+  inline def rightSubstEq: Byte = 24
+  inline def leftSubstIff: Byte = 25
+  inline def rightSubstIff: Byte = 26
+  inline def instSchema: Byte = 27
+  inline def scSubproof: Byte = 28
+  inline def sorry: Byte = 29
```

Comment:
Wouldn't it be better to have a `toByte` with pattern matching? That way we have pattern match exhaustivity checking?